docs/csharp/codedoc.md

```diff
@@ -12,16 +12,464 @@ ms.devlang: dotnet
+        {
+
+        }
+    }
+
+<a name="remarks"></a>
+## &lt;remarks&gt;
+
+The &lt;remarks&gt; tag is used to add information about a type, supplementing the information specified with &lt;summary&gt;.
+
+    /// <summary>
+    /// This class primarily does something
+    /// </summary>
+    /// <remarks>
+    /// It can also do everything
+    /// </remarks>
+    public class SomeClass
+    {
+        
+    }
+
+<a name="returns"></a>
+## &lt;returns&gt;
+
+The &lt;returns&gt; tag should be used in the comment for a method declaration to describe the return value.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <returns>Stuff it has done</returns>
+        public int DoStuff(int a)
+        {
+            return a;
+        }
+    }
+
+<a name="example"></a>
+## &lt;example&gt;
+
+The &lt;example&gt; tag lets you specify an example of how to use a method or other library member. 
+This commonly involves using the [&lt;code&gt;](#code) tag.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <example>
+        /// <code>
+        /// SomeClass someClass = new SomeClass();
+        /// someClass.DoStuff();
+        /// </code>
+        /// </example>
+        public void DoStuff()
+        {
+
+        }
+    }
+
+<a name="value"></a>
+## &lt;value&gt;
+
+The &lt;value&gt; lets you describe the value that a property represents.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <value>The Name property gets/sets the name value of something</value>
+        public string Name { get; set; }
+    }
+
+<a name="exception"></a>
+## &lt;exception&gt;
+
+The &lt;exception&gt; tag lets you specify possible exceptions that can be thrown. 
+It's applicable to methods, properties, events, and indexers.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <exception cref="System.Exception">Thrown when there's no stuff to do</exception>
+        public void DoStuff()
+        {
+
+        }
+    }
+
+The `cref` parameter represents a reference to an exception that is available from the current compilation environment.
+
+<a name="c"></a>
+## &lt;c&gt;
+
+The &lt;c&gt; tag is used to indicate that a single line of text should be marked as code
+
+    /// <summary>
+    /// <c>SomeClass</c> does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <summary>
+        /// <c>DoStuff</c> does stuff on an instance of <c>SomeClass</c>
+        /// </summary>
+        public void DoStuff()
+        {
+
+        }
+    }
+
+<a name="code"></a>
+## &lt;code&gt;
+
+The &lt;code&gt; tag is used to indicate that multiple lines of text should be marked as code.
+See the [&lt;example&gt;](#example) tag for sample usage.
```

Comment:
I'd add that the <code> tag preserves line breaks and indentation for longer examples.